### PR TITLE
Chore: Update async-h1 and related dependencies

### PR DESCRIPTION
**Pull request by kantai**
It looks like #2990 was caused by a bug in an dependency (used by the event observer, so it's only being invoked on configured listeners).

This PR updates the relevant dependencies, which required a few lines of code changes as well.

### CODE REVIEW

File: testnet/puppet-chain/src/main.rs [hunk 1]
```diff
@@ -181,8 +181,8 @@ async fn main() -> http_types::Result<()> {
 }
 
 // Take a TCP stream, and convert it into sequential HTTP request / response pairs.
-async fn accept(addr: String, stream: TcpStream, config: &ConfigFile) -> http_types::Result<()> {
-    async_h1::accept(&addr, stream.clone(), |mut req| async {
+async fn accept(_addr: String, stream: TcpStream, config: &ConfigFile) -> http_types::Result<()> {
```
Comment (pavitthrap) [inline testnet/puppet-chain/src/main.rs:184]: Could you refactor to remove this argument altogether?

**Review by pavitthrap — DISMISSED** [review]
lgtm - just one nit

**Comment by kantai** [comment]
Hmmm -- it looks like updating the dependency is not as simple as I initially thought... will have to look into this a bit more.

**Comment by jcnelson** [comment]
Is this something that we're going to try and get into 2.05.0.1.0?

**Comment by kantai** [comment]
> Is this something that we're going to try and get into 2.05.0.1.0?

That depends on the priority of fixing #2990 I suppose.

This is also something that could go in a patch release.

**Review by gregorycoppola — APPROVED** [review]

**Comment by kantai** [comment]
Looks like all it needed was an update of the tokio version (only used in the integration tests for the integration event observer). Merging!

**Pull request merged**